server.js

```diff
@@ -4,26 +4,85 @@ import cors from 'cors'
 import mongoose from 'mongoose'
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true, useFindAndModify: false })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+const Thought = mongoose.model('Thought', {
+  message: {
+    type: String,
+    required: true,
+    minlength: 5,
+    maxlength: 140
+  },
+  hearts: {
+    type: Number,
+    default: 0
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now
+  }
+})
+
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+app.get('/', async (req, res) => {
+  const sortQuery = req.query.sort
+
+  const sortByQuery = () => {
+    switch (sortQuery) {
+      case 'new':
+        return { createdAt: 'desc' }
+        break
+      case 'old':
+        return { createdAt: 'asc' }
+        break
+      case 'popular':
+        return { hearts: 'desc' }
+        break
+      default:
+        return { createdAt: 'desc' }
+    }
+  }
+  try {
+    const thoughts = await Thought.find().sort(sortByQuery()).limit(20).exec()
+    res.json(thoughts)
+  } catch (err) {
+    res.status(400).json({ message: 'Could not find thoughts', error: err.message })
+  }
+})
+
+app.post('/', async (req, res) => {
+  const { message } = req.body
+  const newThought = new Thought({ message })
+
+  try {
+    await newThought.save()
+    res.status(201).json(newThought)
+  } catch (err) {
+    res.status(400).json({ message: 'Could not save thought', error: err.errors })
+  }
+})
+
+app.post('/:thoughtId/like', async (req, res) => {
+  const thoughtId = req.params.thoughtId
```

Comment:
you can destructure here as well, to be consistent.
`const { thoughtId } = req.params`

Reply:
Thanks, I've changed that now.